examples/linear_model/plot_ridge_coeffs.py

```diff
@@ -1,89 +1,272 @@
-ax.plot(alphas, coefs)
-ax.set_xscale("log")
-plt.xlabel("alpha")
-plt.ylabel("weights")
-plt.title("Ridge coefficients as a function of the regularization")
-plt.axis("tight")
-
-plt.subplot(122)
-ax = plt.gca()
-ax.plot(alphas, errors)
-ax.set_xscale("log")
-plt.xlabel("alpha")
-plt.ylabel("error")
-plt.title("Coefficient error as a function of the regularization")
-plt.axis("tight")
-
-plt.show()
+    errors_coefs.append(mean_squared_error(clf.coef_, w))
+
+# %%
+# Note: In a real setting you should use :class:`~sklearn.linear_model.RidgeCV`
+# for training a Ridge regression with built-in cross-validation, but for the
+# purpose of demonstration we will keep it simple here.
+
+# %%
+# Plotting trained Coefficients and Mean Squared Errors
+#######################################################
+# Like in the first example, we now plot the 10 different regularized
+# coefficients as a function of the regularization parameter `alpha` where each
+# color represents a different coefficient.
+#
+# On the right-hand-side, we plot how the errors of the coefficients from the
+# estimator change as a function of regularization.
+alphas = pd.Index(alphas, name="alpha")
+coefs = pd.DataFrame(coefs, index=alphas, columns=[f"Feature {i}" for i in range(10)])
+errors = pd.Series(errors_coefs, index=alphas, name="Mean squared error")
+
+fig, axs = plt.subplots(1, 2, figsize=(20, 6))
+
+coefs.plot(
+    ax=axs[0],
+    logx=True,
+    ylabel="Ridge coefficient values",
+    title="Ridge coefficients as a function of the regularization strength",
+)
+_ = errors.plot(
+    ax=axs[1],
+    logx=True,
+    ylabel="Mean squared error",
+    title="Coefficient error as a function of the regularization strength",
+)
+
+# %%
+# Interpreting the plots
+########################
+# The plot on the left-hand side shows how the regularization strength (`alpha`)
+# affects the Ridge regression coefficients. Smaller values of `alpha` (weak
+# regularization), don't affect the trained coefficients very much. Comparing
+# those with the true coefficients `w` from the noisy data set however reveals
+# that the model hardly captures the true weights. As `alpha` increases, the
+# coefficients shrink towards zero.
+#
+# The right-hand side plot shows the `mean squared error` between the
+# coefficients found by the model and the true coefficients (`w`). It provides a
+# measure that relates to how exact our ridge model is in comparison to the true
+# generative model. A low error means that it found coefficients closer to the
+# ones of the true generative model. In this case, we can see that the curve has
+# a minimum at `alpha = 10 ** 1.7` (about 50), where the `mean squared error` is
+# lowest. This means, that this particular regularization strength recovers the
+# true coefficients best. A higher regularization however would cause the model
+# to underfit.
+
+# %%
+# Conclusion
+# ----------
+# The four plots visualize the trade-off between fitting the training data too
```

Comment:
So we need to reformulate the conclusion for the two plots that you have for the moment.